code/modules/spells/spell_types/vampire_powers.dm

```diff
@@ -0,0 +1,416 @@
+		new /obj/effect/decal/remains/human(L.loc)
+		L.dust()
+	to_chat(L, "<span class='notice'>You begin to reanimate... this will take a minute.</span>")
+	addtimer(CALLBACK(src, /obj/effect/proc_holder/spell/self/revive.proc/revive, L), 600)
+
+/obj/effect/proc_holder/spell/self/revive/proc/revive(mob/living/user)
+	user.revive(full_heal = TRUE)
+	user.visible_message("<span class='warning'>[user] reanimates from death!</span>", "<span class='notice'>We get back up.</span>")
+	playsound(user, 'sound/magic/demon_consume.ogg', 50, 1)
+	var/list/missing = user.get_missing_limbs()
+	if(missing.len)
+		user.visible_message("<span class='warning'>Shadowy matter takes the place of [user]'s missing limbs as they reform!</span>")
+		user.regenerate_limbs(0, list(BODY_ZONE_HEAD))
+	user.regenerate_organs()
+	user.adjustStaminaLoss(100) //Dont be ready to skeddadle without atleast without having to use some additional blood.
+
+/obj/effect/proc_holder/spell/self/summon_coat
+	name = "Summon Vampire Coat (100)"
+	gain_desc = "Now that you have reached full power, you can now pull a vampiric coat out of thin air!"
+	desc = "Summons a vampiric coat with decent armor, with a hood that hides your face."
+	blood_used = 100
+	action_icon = 'icons/mob/vampire.dmi'
+	action_icon_state = "coat"
+	action_background_icon_state = "bg_demon"
+	vamp_req = TRUE
+
+/obj/effect/proc_holder/spell/self/summon_coat/cast(list/targets, mob/user = usr)
+	if(!is_vampire(user) || !isliving(user))
+		revert_cast()
+		return
+	var/datum/antagonist/vampire/V = user.mind.has_antag_datum(/datum/antagonist/vampire)
+	if(!V)
+		return
+	if(QDELETED(V.coat) || !V.coat)
+		V.coat = new /obj/item/clothing/suit/hooded/vamp_coat(user.loc)
+	else if(get_dist(V.coat, user) > 1 || !(V.coat in user.GetAllContents()))
+		V.coat.forceMove(user.loc)
+	user.put_in_hands(V.coat)
+	to_chat(user, "<span class='notice'>You summon your vampire coat.</span>")
+
+
+/obj/effect/proc_holder/spell/self/batform
+	name = "Bat Form (15)"
+	gain_desc = "You now have the Bat Form ability, which allows you to turn into a bat (and back!)"
+	desc = "Transform into a bat!"
+	action_icon_state = "bat"
+	charge_max = 200
+	blood_used = 0 //this is only 0 so we can do our own custom checks
+	action_icon = 'icons/mob/vampire.dmi'
+	action_background_icon_state = "bg_demon"
+	vamp_req = TRUE
+	var/mob/living/simple_animal/hostile/retaliate/bat/vampire_bat/bat
+
+/obj/effect/proc_holder/spell/self/batform/cast(list/targets, mob/user = usr)
+	var/datum/antagonist/vampire/V = user.mind.has_antag_datum(/datum/antagonist/vampire)
+	if(!V)
+		return FALSE
+	if(!bat || bat.stat == DEAD)
+		if(V.usable_blood < 15)
+			to_chat(user, "<span class='warning'>You do not have enough blood to cast this!</span>")
+			return FALSE
+		bat = new /mob/living/simple_animal/hostile/retaliate/bat/vampire_bat(user.loc)
+		user.forceMove(bat)
+		bat.controller = user
+		user.status_flags |= GODMODE
```

Comment:
Puts the user inside the bat right? And not the bat itself has the godmode?

Reply:
this needs to be subtyped to shapeshift at some point